crates/build/src/lib.rs

```diff
@@ -119,14 +126,56 @@ pub fn build_program(path: &str) {
 ///
 /// # Arguments
 ///
-/// * `path` - A string slice that holds the path to the program directory.
+/// * `path` - A path to the guest program directory
+///
 /// * `args` - A [`BuildArgs`] struct that contains various build configuration options.
 ///
 /// Set the `SP1_SKIP_PROGRAM_BUILD` environment variable to `true` to skip building the program.
-pub fn build_program_with_args(path: &str, args: BuildArgs) {
+///
+/// ## Note: Using this function without an absolute path is not recommended.
+/// Try using the `build_program_from_path!` macro instead.
+pub fn build_program_with_args(path: impl AsRef<Path>, args: BuildArgs) {
     build_program_internal(path, Some(args))
 }
 
+/// Builds the program with the given arguments if the program at path, or one of its dependencies,
+///
+/// ### Note:
+/// This function is only exposed to support the `build_program_from_path!` macro.
+/// It is not recommended to use this function directly.
+pub fn build_program_with_maybe_args(path: impl AsRef<Path>, args: Option<BuildArgs>) {
+    build_program_internal(path, args)
+}
+
+/// Build a program at the given path.
+///
+/// # Arguments
+/// * `path` - A path to the guest program directory, if not absolute, assumed to be relative to
+///           the callers manifest directory.
+///
+///   `args` - A [`BuildArgs`] struct that contains various build configuration options.
+///            If not provided, the default options are used.
+#[macro_export]
+macro_rules! build_program_from_path {
```

Comment:
wait why does this have to be a macro?

Reply:
Because we need to inline the manifest dir at compile time so the path is always relative from the crate dir

from slack: 
My contrived example of why this is bad is:
Crate A: lives in dir A exports a function build_A_program that uses a relative path to call sp1_build::build_program
Crate B:  lives in dir B imports this function
Crate B: Calls build_A_program with working dir B
this call will then try to look for the path relative to B which is def incorrect